src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateStore.java

```diff
@@ -142,15 +141,24 @@ public PrivateKey getCAPrivateKey() {
      * @throws KeyStoreException if unable to retrieve the certificate
      */
     public X509Certificate getCACertificate() throws KeyStoreException {
-        return (X509Certificate) getCaCertificateChain()[0];
+        return getCaCertificateChain()[0];
+    }
+
+    private void setCaCertificateChain(Certificate... caCertificateChain) throws KeyStoreException {
```

Comment:
do we want to handle case if chain is null?  probably not happening in practice but may be good to have for completeness

Reply:
It would be nice if the type system could tell it we don't accept null values, would that @NotNull annotation work?

Reply:
It does, but it just auto-generates the code that throws a NPE. I think in this case we prefer a checked exception. Will add